subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java

```diff
@@ -334,8 +337,13 @@ public M getData() {
                     throw new UnsupportedOperationException("Cannot load script that does nothing.");
                 }
                 try {
-                    // Classloader scope will be handled by the cache, class will be released when the classloader is.
-                    ClassLoader loader = classLoaderCache.put(classLoaderId, new ScriptClassLoader(source, classLoader, DefaultClassPath.of(scriptCacheDir), sourceHashCode));
+                    ClassPath scriptClassPath = DefaultClassPath.of(scriptCacheDir);
+                    // Create scope for recording purpose only
```

Comment:
Perhaps add an instant execution TODO to use the scope to create the ClassLoader, as we do elsewhere, rather than creating an ad hoc ClassLoader and (not-quite-correctly) connecting the pieces together.

I probably wouldn't fix this yet (but we could).

Reply:
Done